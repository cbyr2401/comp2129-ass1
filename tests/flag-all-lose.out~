g 10 10
b 0 1
b 0 2
b 0 3
b 0 4
b 0 5
b 0 6
b 0 7
b 1 1
b 2 2
b 3 3
+----------+
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 0
+----------+
|f*********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 1
+----------+
|f*********|
|f*********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 2
+----------+
|f*********|
|f*********|
|f*********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 3
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 4
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|**********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 5
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 6
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|**********|
|**********|
|**********|
+----------+
f 0 7
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|**********|
|**********|
+----------+
f 0 8
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|**********|
+----------+
f 0 9
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
+----------+
f 1 0
+----------+
|ff********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
+----------+
f 1 1
+----------+
|ff********|
|ff********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
+----------+
f 1 2
+----------+
|ff********|
|ff********|
|ff********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
+----------+
f 1 3
+----------+
|ff********|
|ff********|
|ff********|
|ff********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
+----------+
f 1 4
+----------+
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
+----------+
f 1 5
+----------+
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|f*********|
|f*********|
|f*********|
|f*********|
+----------+
f 1 6
+----------+
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|f*********|
|f*********|
|f*********|
+----------+
f 1 7
+----------+
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|f*********|
|f*********|
+----------+
f 1 8
+----------+
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|f*********|
+----------+
f 1 9
+----------+
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
+----------+
f 2 0
+----------+
|fff*******|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
+----------+
f 2 1
+----------+
|fff*******|
|fff*******|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
+----------+
f 2 2
+----------+
|fff*******|
|fff*******|
|fff*******|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
+----------+
f 2 3
+----------+
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
+----------+
f 2 4
+----------+
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|ff********|
|ff********|
|ff********|
|ff********|
|ff********|
+----------+
f 2 5
+----------+
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|ff********|
|ff********|
|ff********|
|ff********|
+----------+
f 2 6
+----------+
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|ff********|
|ff********|
|ff********|
+----------+
f 2 7
+----------+
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|ff********|
|ff********|
+----------+
f 2 8
+----------+
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|ff********|
+----------+
f 2 9
+----------+
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
+----------+
f 3 0
+----------+
|ffff******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
+----------+
f 3 1
+----------+
|ffff******|
|ffff******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
+----------+
f 3 2
+----------+
|ffff******|
|ffff******|
|ffff******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
+----------+
f 3 3
+----------+
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
+----------+
f 3 4
+----------+
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
+----------+
f 3 5
+----------+
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|fff*******|
|fff*******|
|fff*******|
|fff*******|
+----------+
f 3 6
+----------+
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|fff*******|
|fff*******|
|fff*******|
+----------+
f 3 7
+----------+
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|fff*******|
|fff*******|
+----------+
f 3 8
+----------+
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|fff*******|
+----------+
f 3 9
+----------+
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
+----------+
f 4 0
+----------+
|fffff*****|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
+----------+
f 4 1
+----------+
|fffff*****|
|fffff*****|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
+----------+
f 4 2
+----------+
|fffff*****|
|fffff*****|
|fffff*****|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
+----------+
f 4 3
+----------+
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
+----------+
f 4 4
+----------+
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
+----------+
f 4 5
+----------+
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|ffff******|
|ffff******|
|ffff******|
|ffff******|
+----------+
f 4 6
+----------+
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|ffff******|
|ffff******|
|ffff******|
+----------+
f 4 7
+----------+
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|ffff******|
|ffff******|
+----------+
f 4 8
+----------+
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|ffff******|
+----------+
f 4 9
+----------+
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
+----------+
f 5 0
+----------+
|ffffff****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
+----------+
f 5 1
+----------+
|ffffff****|
|ffffff****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
+----------+
f 5 2
+----------+
|ffffff****|
|ffffff****|
|ffffff****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
+----------+
f 5 3
+----------+
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
+----------+
f 5 4
+----------+
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
+----------+
f 5 5
+----------+
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|fffff*****|
|fffff*****|
|fffff*****|
|fffff*****|
+----------+
f 5 6
+----------+
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|fffff*****|
|fffff*****|
|fffff*****|
+----------+
f 5 7
+----------+
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|fffff*****|
|fffff*****|
+----------+
f 5 8
+----------+
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|fffff*****|
+----------+
f 5 9
+----------+
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
+----------+
f 6 0
+----------+
|fffffff***|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
+----------+
f 6 1
+----------+
|fffffff***|
|fffffff***|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
+----------+
f 6 2
+----------+
|fffffff***|
|fffffff***|
|fffffff***|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
+----------+
f 6 3
+----------+
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
+----------+
f 6 4
+----------+
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
+----------+
f 6 5
+----------+
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|ffffff****|
|ffffff****|
|ffffff****|
|ffffff****|
+----------+
f 6 6
+----------+
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|ffffff****|
|ffffff****|
|ffffff****|
+----------+
f 6 7
+----------+
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|ffffff****|
|ffffff****|
+----------+
f 6 8
+----------+
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|ffffff****|
+----------+
f 6 9
+----------+
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
+----------+
f 7 0
+----------+
|ffffffff**|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
+----------+
f 7 1
+----------+
|ffffffff**|
|ffffffff**|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
+----------+
f 7 2
+----------+
|ffffffff**|
|ffffffff**|
|ffffffff**|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
+----------+
f 7 3
+----------+
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
+----------+
f 7 4
+----------+
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
+----------+
f 7 5
+----------+
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|fffffff***|
|fffffff***|
|fffffff***|
|fffffff***|
+----------+
f 7 6
+----------+
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|fffffff***|
|fffffff***|
|fffffff***|
+----------+
f 7 7
+----------+
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|fffffff***|
|fffffff***|
+----------+
f 7 8
+----------+
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|fffffff***|
+----------+
f 7 9
+----------+
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
+----------+
f 8 0
+----------+
|fffffffff*|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
+----------+
f 8 1
+----------+
|fffffffff*|
|fffffffff*|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
+----------+
f 8 2
+----------+
|fffffffff*|
|fffffffff*|
|fffffffff*|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
+----------+
f 8 3
+----------+
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
+----------+
f 8 4
+----------+
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
+----------+
f 8 5
+----------+
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|ffffffff**|
|ffffffff**|
|ffffffff**|
|ffffffff**|
+----------+
f 8 6
+----------+
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|ffffffff**|
|ffffffff**|
|ffffffff**|
+----------+
f 8 7
+----------+
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|ffffffff**|
|ffffffff**|
+----------+
f 8 8
+----------+
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|ffffffff**|
+----------+
f 8 9
+----------+
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
+----------+
f 9 0
+----------+
|ffffffffff|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
+----------+
f 9 1
+----------+
|ffffffffff|
|ffffffffff|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
+----------+
f 9 2
+----------+
|ffffffffff|
|ffffffffff|
|ffffffffff|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
+----------+
f 9 3
+----------+
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
+----------+
f 9 4
+----------+
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
+----------+
f 9 5
+----------+
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|fffffffff*|
|fffffffff*|
|fffffffff*|
|fffffffff*|
+----------+
f 9 6
+----------+
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|fffffffff*|
|fffffffff*|
|fffffffff*|
+----------+
f 9 7
+----------+
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|fffffffff*|
|fffffffff*|
+----------+
f 9 8
+----------+
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|fffffffff*|
+----------+
f 9 9
+----------+
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
|ffffffffff|
+----------+
lost
