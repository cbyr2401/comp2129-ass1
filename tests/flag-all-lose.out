g 10 10
b 0 1
b 0 2
b 0 3
b 0 4
b 0 5
b 0 6
b 0 7
b 1 1
b 2 2
b 3 3
+----------+
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 0
+----------+
|f*********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 1
+----------+
|f*********|
|f*********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 2
+----------+
|f*********|
|f*********|
|f*********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 3
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|**********|
|**********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 4
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|**********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 5
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|**********|
|**********|
|**********|
|**********|
+----------+
f 0 6
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|**********|
|**********|
|**********|
+----------+
f 0 7
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|**********|
|**********|
+----------+
f 0 8
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|**********|
+----------+
f 0 9
+----------+
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
|f*********|
+----------+
error
